### PR TITLE
[PATCH v3] Merge ODP linux-generic v1.42.0.0

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: platform/linux-dpdk/odp_time.c [hunk 1]
```diff
@@ -1,302 +1,45 @@
...
-	return false;
-}
-
 int _odp_time_init_global(void)
 {
-	struct timespec *timespec;
-	int ret = 0;
 	_odp_time_global_t *global = &_odp_time_glob;
 
 	memset(global, 0, sizeof(_odp_time_global_t));
```
Comment (TuomasTaipale) [inline platform/linux-dpdk/odp_time.c:28]: `_odp_time_glob` could be used here directly.

File: platform/linux-dpdk/odp_time.c [hunk 2]
```diff
@@ -1,302 +1,45 @@
...
-		global->use_hw = 1;
-		global->hw_freq_hz  = _odp_cpu_global_time_freq();
-
-		if (global->hw_freq_hz == 0)
-			return -1;
-
-		_ODP_PRINT("HW time counter freq: %" PRIu64 " hz\n\n", global->hw_freq_hz);
+#ifdef RTE_LIBEAL_USE_HPET
+	if (rte_eal_hpet_init(1) != 0)
+		_ODP_ERR("HPET init failed. Using TSC time.\n");
```
Comment (TuomasTaipale) [inline platform/linux-dpdk/odp_time.c:32]: Maybe a `_ODP_PRINT` or `_ODP_WARN` as this does not lead to an error to be returned?

File: platform/linux-dpdk/odp_time.c (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-dpdk/odp_time.c:28]: Fixed in V2.

File: platform/linux-dpdk/odp_time.c (hunk 2 above)
Comment (MatiasElo) [reply platform/linux-dpdk/odp_time.c:32]: True, changed to warning in V2.

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**